src/sample_weight_audit/estimator_check.py

```diff
@@ -172,12 +172,34 @@ def compute_predictions(est, X):
             return est.predict_proba(X)
         else:
             return est.decision_function(X)
+    elif isinstance(est, Pipeline):
+        return est.predict(X)
     elif hasattr(est, "transform"):
         return est.transform(X)
     else:
         raise NotImplementedError(f"Estimator type not supported: {est}")
 
 
+def check_pipeline_and_fit(est, X, y, sample_weight=None, seed=None):
+
+    if hasattr(est, "transform"):
```

Comment:
Some transformers are directly classifiers (e.g. `LinearDiscriminantAnalysis`) so we don't need to wrap them. This might also be the case for some dual transformer/regressors, although I am not sure.

Reply:
Similarly, in the future, KMeans would be better be evaluated directly as a clusterer (with a clustering metric) than a transformer, but we can keep clusterer outside the scope of this PR for now.